pkg/controller/association/conf.go

```diff
@@ -103,6 +107,38 @@ func ElasticsearchAuthSettings(c k8s.Client, association commonv1.Association) (
 	return assocConf.AuthSecretKey, string(data), nil
 }
 
+// AllowVersion returns true if the given resourceVersion is lower or equal to the associations version.
+// For example: Kibana in version 7.8.0 cannot be deployed if its Elasticsearch association reports version 7.7.0.
+// Referenced resources version is parsed from the association conf annotation.
+func AllowVersion(resourceVersion version.Version, associated commonv1.Associated, logger logr.Logger) bool {
```

Comment:
Hm same as https://github.com/elastic/cloud-on-k8s/pull/3537 I would consider this to be an improvement/fix about something that should not have worked initially.

Reply:
> Hm same as #3537 I would consider this to be an improvement/fix about something that should not have worked initially. 

You linked to this PR :) 
It seems to me that this is a technicality and while I agree we could argue that this is a bug fix and this was always the intended behaviour the user experience will be bad. 

As soon as this is merged and without being marked as `>breaking` it's no longer discoverable in any way. So user that's running Kibana/APM higher than ES has changes to Kibana/APM effectively disabled (including config, cert, securesettings changes) until they either rollback ECK (I'm not even sure if this just works) or update ES (might not be possible at all for their own reasons). 

I think it doesn't cost us anything to document this change, but it can damage the experience for some users significantly.

Reply:
👍  let's mark this as breaking.